siac/walletcmd.go

```diff
@@ -206,22 +238,86 @@ func walletsiafundssendcmd(cmd *cobra.Command, args []string) {
 	*/
 }
 
+// walletstatuscmd retrieves and displays information about the wallet
 func walletstatuscmd() {
-	/*
-			status := new(modules.WalletInfo)
-			err := getAPI("/wallet/status", status)
-			if err != nil {
-				fmt.Println("Could not get wallet status:", err)
-				return
-			}
-			// divide by 1e24 to get SC
-			r := new(big.Rat).SetFrac(status.Balance.Big(), new(big.Int).Exp(big.NewInt(10), big.NewInt(24), nil))
-			sc, _ := r.Float64()
-			fmt.Printf(`Wallet status:
-		Balance:   %.2f SC
-		Exact:     %v H
-		Addresses: %d
-		`, sc, status.Balance, status.NumAddresses)
-	*/
-	fmt.Println("wallet status not implemented in siac")
+	status := new(api.WalletGET)
+	err := getAPI("/wallet", status)
+	if err != nil {
+		fmt.Println("Could not get wallet status:", err)
+		return
+	}
+	encStatus := "Unencrypted"
+	if status.Encrypted {
+		encStatus = "Encrypted"
+	}
+	lockStatus := "Locked"
+	if status.Unlocked {
+		lockStatus = "Unlocked"
+	}
+	// divide by 1e24 to get SC
+	r := new(big.Rat).SetFrac(status.ConfirmedSiacoinBalance.Big(), new(big.Int).Exp(big.NewInt(10), big.NewInt(24), nil))
+	sc, _ := r.Float64()
+	fmt.Printf(`Wallet status:
+%s, %s
+Balance:   %.2f SC
+Exact:     %v H
+`, encStatus, lockStatus, sc, status.ConfirmedSiacoinBalance)
+}
+
+// walletunlockcmd unlocks a saved wallet
+func walletunlockcmd(password string) {
+	qs := fmt.Sprintf("encryptionpassword=%s&dictonary=%s", password, "english")
+	err := post("/wallet/unlock", qs)
+	if err != nil {
+		fmt.Println("Could not unlock wallet:", err)
+		return
+	}
+	fmt.Println("Wallet unlocked")
+}
+
+// walletencryptcmd encrypts the wallet with the given password
+func walletinitcmd() {
+	var er api.WalletEncryptPOST
+	qs := fmt.Sprintf("dictionary=%s", "english")
+	err := postResp("/wallet/encrypt", qs, &er)
+	if err != nil {
+		fmt.Println("Error when encrypting wallet:", err)
+		return
+	}
+	fmt.Printf("Encrypted wallet with primary seed: %s\n", er.PrimarySeed)
```

Comment:
should init take an argument then? I think it would be easier to just always use the seed as the password. Although that means your password is pretty long...